### ISSUE TITLE
It's possible to pass days in `yearsAndMonths duration` variable

### ISSUE THREAD

**Issue by yesamer** [apache/incubator-kie#7068]
The engine can parse, execute, and manage `years and month duration` values like `P1Y1M1D`.

<img width="689" alt="Screenshot 2024-05-21 at 18 07 30" src="https://github.com/apache/incubator-kie/assets/16005046/a893750f-b8c8-4480-b331-2ff9772b1248">

As the type name suggests, that value should be invalid. 
The reason for that behavior is the engine manages the `years and month duration` type as `Period` Java type, which allows the days' usage.

@baldimir @gitgabrio

**Comment by baldimir** [comment]
Based on testing, looks like the value of days is ignored. I added this test to FEELDateTimeDurationTest and it passes. 

` { "@\"P2Y2M1D\"", ComparablePeriod.parse( "P2Y2M" ) , null},`

**Comment by baldimir** [comment]
To clarify - the engine does not use Period directly. It wraps it in a custom class [ComparablePeriod](https://github.com/apache/incubator-kie-drools/blob/18047e93c81c3502cd759fbe3e8ce0c8f2e796f4/kie-dmn/kie-dmn-feel/src/main/java/org/kie/dmn/feel/lang/types/impl/ComparablePeriod.java#L33).

**Comment by yesamer** [comment]
Yes I saw it, but the actual Duration value is stored in a `Period`
